core/src/main/scala/cats/data/OptionT.scala

```diff
@@ -248,7 +248,28 @@ sealed abstract private[data] class OptionTInstances extends OptionTInstances0 {
         OptionT(F.defer(fa.value))
     }
 
-  implicit def catsDateTraverseFilterForOptionT[F[_]](implicit F0: Traverse[F]): TraverseFilter[OptionT[F, *]] =
+  implicit def catsDataTraverseFilterForOptionT[F[_]](implicit F0: Traverse[F]): TraverseFilter[OptionT[F, *]] =
+    new OptionTFunctorFilter[F] with TraverseFilter[OptionT[F, *]] {
+      implicit def F: Functor[F] = F0
+
+      val traverse: Traverse[OptionT[F, *]] = OptionT.catsDataTraverseForOptionT[F]
+
+      def traverseFilter[G[_], A, B](
+        fa: OptionT[F, A]
+      )(f: A => G[Option[B]])(implicit G: Applicative[G]): G[OptionT[F, B]] =
+        G.map(Traverse[F].traverse[G, Option[A], Option[B]](fa.value) { oa =>
+          TraverseFilter[Option].traverseFilter(oa)(f)
+        })(OptionT[F, B])
+
+      override def filterA[G[_], A](
+        fa: OptionT[F, A]
+      )(f: A => G[Boolean])(implicit G: Applicative[G]): G[OptionT[F, A]] =
+        G.map(Traverse[F].traverse(fa.value)(TraverseFilter[Option].filterA[G, A](_)(f)))(OptionT[F, A])
+
+    }
+
+  @deprecated("renamed to catsDataTraverseFilterForOptionT", "2.0.0")
+  def catsDateTraverseFilterForOptionT[F[_]](implicit F0: Traverse[F]): TraverseFilter[OptionT[F, *]] =
```

Comment:
I think this could call the corrected one instead of duplicating the implementation. The key is that it's deprecated and no longer the implicit.